onnxruntime/core/providers/cuda/nn/conv_transpose.cc

```diff
@@ -101,9 +116,18 @@ Status ConvTranspose<T, NHWC>::DoConvTranspose(OpKernelContext* context, bool dy
   bool has_bias = dynamic_padding ? num_inputs == 4 : num_inputs == 3;
 
   CudaT* y_data = nullptr;
+
+  // convert 1D to 2D
   if (x_dimensions == 3) {
-    x_dims.insert(x_dims.begin() + 2, 1);
-    w_dims.insert(w_dims.begin() + 2, 1);
+    const auto insert_at = NHWC ? 1 : 2;
+
+    // NCHW: N, C, k1, C -> N, C, 1, k1
+    // NHWC: N, k1, C -> N, 1, k1, C
```

Comment:
By default, we have been padding the fake feature dim to the right of the existing feature dim - https://github.com/microsoft/onnxruntime/blob/8fbb8a149f7886ce17460a051a9d9ae66fd53a6a/onnxruntime/core/providers/cuda/nn/conv.cc#L268. Can we keep it consistent unless this padding gets best performance from Cudnn for some reason ?